### PR TITLE
[DO NOT MERGE] PLANNER-937: Added development documentation

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
@MusaTalluzi Incomplete, documenting the code architecture of this project; still need to do Roster Screens and Server.

### CODE REVIEW

File: optashift-employee-rostering-docs/src/main/asciidoc/Development/Client/Client-section.adoc [hunk 1]
```diff
@@ -0,0 +1,171 @@
+[[clientDevelopment]]
+= Client Side Development of Optashift Employee Rostering
+:imagesdir: ../..
+
+== Basics
+
+Optashift Employee Rostering is a kiegroup project, and as such you should
```
Comment (ge0ffrey) [inline optashift-employee-rostering-docs/src/main/asciidoc/Development/Client/Client-section.adoc]: OptaWeb (w is capital)

File: optashift-employee-rostering-docs/src/main/asciidoc/Development/Client/Client-section.adoc [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+* `deleteRow(Model row)`, which deletes _row_ from the database by calling the corresponding REST method.
+  Should fire `DataInvalidation<Model>` on success or failure.
+* `updateRow(Model oldRow, Model newRow)`, which updates _oldRow_ to _newRow_ in the database by calling the corresponding REST method.
+  Should fire `DataInvalidation<Model>` on success or failure.
+* `createRow(Model row)`, which creates _row_ in the database by calling the corresponding REST method.
+  Should fire `DataInvalidation<Model>` on success or failure.
+* An `@PostConstruct` method, which binds the model to the row by calling `dataBinder.bind(<editorWidgetForField>, "<modelField>")`
+  for each editable field in the model.
+
+To add a new row to the table, you can call `ModelRow.createNewRow(emptyValue, table, pager)`,
```
Comment (ge0ffrey) [inline optashift-employee-rostering-docs/src/main/asciidoc/Development/Client/Client-section.adoc]: Use active voice: "you can call" -> "call"

File: optashift-employee-rostering-docs/src/main/asciidoc/Development/Server/Server-section.adoc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[[serverDevelopment]]
+= Server Side Development of Optashift Employee Rostering
+:imagesdir: ../..
+
+TODO
```
Comment (ge0ffrey) [inline optashift-employee-rostering-docs/src/main/asciidoc/Development/Server/Server-section.adoc]: TODO's in documentation look bad and amateuristic. There are always documentation sections left to do. Docs are never perfect, never complete.

just leave it out the section are aren't documented yet (delete them). If they need to be tracked to do, create a jira.

**Comment by Christopher-Chianelli** [comment]
Closed, PLANNER-937 now refers to a different issue

**Pull request closed without merging**